<!doctype html>
<html class="no-js" lang="en" data-content_root="../">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="index" title="Index" href="../genindex.html" /><link rel="search" title="Search" href="../search.html" /><link rel="next" title="Process" href="process.html" /><link rel="prev" title="Theory" href="../theory.html" />

    <!-- Generated with Sphinx 7.2.6 and Furo 2024.01.29 -->
        <title>Download - PPI Origami documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=a746c00c" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo.css?v=135e06be" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo-extensions.css?v=36a5483c" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../index.html"><div class="brand">PPI Origami  documentation</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="../index.html">
  
  <div class="sidebar-logo-container">
    <img class="sidebar-logo" src="../_static/logo.svg" alt="Logo"/>
  </div>
  
  <span class="sidebar-brand-text">PPI Origami  documentation</span>
  
</a><form class="sidebar-search-container" method="get" action="../search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../install.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../usage.html">Usage Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory.html">Theory</a></li>
<li class="toctree-l1 current current-page"><a class="current reference internal" href="#">Download</a></li>
<li class="toctree-l1"><a class="reference internal" href="process.html">Process</a></li>
<li class="toctree-l1"><a class="reference internal" href="analysis.html">Analysis</a></li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          
<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main">
          <section id="download">
<h1>Download<a class="headerlink" href="#download" title="Link to this heading">#</a></h1>
<p>The <code class="docutils literal notranslate"><span class="pre">download</span></code> module of PPI Origami allows you to download files from their authoratative sources. PPI Origami works best when you designate one folder on your filesystem for keeping all original, untransformed datasets (we’ll call that the “raw folder”). You’ll refer to this folder in the <code class="docutils literal notranslate"><span class="pre">process</span></code> module, where “raw” files will be transformed and saved in a “processed” folder.</p>
<p>You can find a description of all the possible download commands by running:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="o">--</span><span class="n">help</span>
</pre></div>
</div>
<p>Information specific to arguments of commands can be found by running the command with the help flag:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">COMMAND</span> <span class="o">--</span><span class="n">help</span>
</pre></div>
</div>
<p>This information is reproduced on this page.</p>
<dl class="py class">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ppi_origami.__main__.</span></span><span class="sig-name descname"><span class="pre">Download</span></span><a class="headerlink" href="#ppi_origami.__main__.Download" title="Link to this definition">#</a></dt>
<dd><dl class="py method">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download.biogrid">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">biogrid</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">version</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'4.4.224'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ppi_origami.__main__.Download.biogrid" title="Link to this definition">#</a></dt>
<dd><p>Download the <a class="reference external" href="https://thebiogrid.org/">BioGRID</a> PPI dataset to the raw folder. Namely, it downloads the <code class="docutils literal notranslate"><span class="pre">BIOGRID-ORGANISM-{version}.mitab.zip</span></code> file from the <a class="reference external" href="https://downloads.thebiogrid.org/BioGRID/Release-Archive/">BioGRID release archive</a>.</p>
<p>You can call this function from the CLI using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">biogrid</span> <span class="n">RAW_FOLDER</span> <span class="o">--</span><span class="n">version</span> <span class="mf">4.4.224</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>raw_folder</strong> (<em>pathlib.Path</em>) – The folder to download the dataset to.</p></li>
<li><p><strong>version</strong> (<em>str</em>) – The BioGRID version to download, defaults to “4.4.224”.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download.dscript">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">dscript</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">taxon</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ppi_origami.__main__.Download.dscript" title="Link to this definition">#</a></dt>
<dd><p>Download a <a class="reference external" href="https://dscript.csail.mit.edu/">D-SCRIPT</a> PPI dataset to the raw folder. D-SCRIPT specifies datasets for <em>H. sapiens</em>, <em>M. musculus</em>, <em>D. melanogaster</em>, <em>S. cerevisiae</em>, <em>C. elegans</em>, and <em>E. coli</em>.</p>
<p>These organisms correspond to the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI Taxon IDs</a>: 9606, 10090, 7227, 4932, 6239, and 511145. These are the only valid values of the <code class="docutils literal notranslate"><span class="pre">taxon</span></code> argument.</p>
<p>You can call this function from the CLI using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">dscript</span> <span class="n">RAW_FOLDER</span> <span class="n">TAXON</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>raw_folder</strong> (<em>pathlib.Path</em>) – The folder to download the dataset to.</p></li>
<li><p><strong>taxon</strong> (<em>int</em>) – The NCBI taxon ID of the organism whose links you wish to download. Must be on of 9606, 10090, 7227, 4932, 6239, or 511145.</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ValueError</strong> – A ValueError is raised when <code class="docutils literal notranslate"><span class="pre">taxon</span></code> is not one of 9606, 10090, 7227, 4932, 6239, or 511145.</p>
</dd>
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download.hippie">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">hippie</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">version</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'current'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ppi_origami.__main__.Download.hippie" title="Link to this definition">#</a></dt>
<dd><p>Download the <a class="reference external" href="https://cbdm-01.zdv.uni-mainz.de/~mschaefer/hippie/">HIPPIE</a> PPI dataset to the raw folder. You may specify the version to download. A value of “current” results in the latest version being downloaded. To download older version, supply a version number like “2.2”.</p>
<p>You can call this function from the CLI using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">hippie</span> <span class="n">RAW_FOLDER</span> <span class="o">--</span><span class="n">version</span> <span class="n">current</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>raw_folder</strong> (<em>pathlib.Path</em>) – The folder to download the dataset to.</p></li>
<li><p><strong>version</strong> (<em>str</em>) – The version of the HIPPIE dataset to download, defaults to “current”.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download.oma">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">oma</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ppi_origami.__main__.Download.oma" title="Link to this definition">#</a></dt>
<dd><p>Download orthology data from <a class="reference external" href="https://omabrowser.org/oma/home/">OMA</a> to the raw folder. Specifically, it downloads the orthology data in a gzipped XML format (<code class="docutils literal notranslate"><span class="pre">oma-groups.orthoXML.xml.gz</span></code>), as well as mappings between OMA identifiers and UniProt identifiers (<code class="docutils literal notranslate"><span class="pre">oma-uniprot.txt.gz</span></code>).</p>
<p>You can call this function from the CLI using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">oma</span> <span class="n">RAW_FOLDER</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>raw_folder</strong> (<em>pathlib.Path</em>) – The folder to download the dataset to.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download.string_aliases">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">string_aliases</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">version</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'12.0'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ppi_origami.__main__.Download.string_aliases" title="Link to this definition">#</a></dt>
<dd><p>Download the <a class="reference external" href="https://string-db.org/">STRING</a> aliases dataset. These are mappings between STRING identifiers, and other identifiers (most importantly, UniProt).</p>
<p>You can call this function from the CLI using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">string_aliases</span> <span class="n">RAW_FOLDER</span> <span class="o">--</span><span class="n">version</span> <span class="mf">12.0</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>raw_folder</strong> (<em>pathlib.Path</em>) – The folder to download the dataset to.</p></li>
<li><p><strong>version</strong> (<em>str</em>) – The version of the STRING database to download.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download.string_links">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">string_links</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">version</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'12.0'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">taxon</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ppi_origami.__main__.Download.string_links" title="Link to this definition">#</a></dt>
<dd><p>Download the <a class="reference external" href="https://string-db.org/">STRING</a> links dataset. Downloads the <code class="docutils literal notranslate"><span class="pre">protein.links.{version}.txt.gz</span></code> file.</p>
<p>You can call this function from the CLI using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">string_links</span> <span class="n">RAW_FOLDER</span> <span class="o">--</span><span class="n">version</span> <span class="mf">12.0</span> <span class="o">--</span><span class="n">taxon</span> <span class="mi">9606</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>raw_folder</strong> (<em>pathlib.Path</em>) – The folder to download the dataset to.</p></li>
<li><p><strong>version</strong> (<em>str</em>) – The version of the STRING database to download.</p></li>
<li><p><strong>taxon</strong> (<em>Optional</em><em>[</em><em>int</em><em>]</em>) – The NCBI taxon ID of the organism whose links you wish to download. Omit for all organisms. Defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download.string_links_detailed">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">string_links_detailed</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">version</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'12.0'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">taxon</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ppi_origami.__main__.Download.string_links_detailed" title="Link to this definition">#</a></dt>
<dd><p>Download the <a class="reference external" href="https://string-db.org/">STRING</a> links dataset. Downloads the <code class="docutils literal notranslate"><span class="pre">protein.links.detailed.{version}.txt.gz</span></code> file.</p>
<p>You can call this function from the CLI using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">string_links_detailed</span> <span class="n">RAW_FOLDER</span> <span class="o">--</span><span class="n">version</span> <span class="mf">12.0</span> <span class="o">--</span><span class="n">taxon</span> <span class="mi">9606</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>raw_folder</strong> (<em>pathlib.Path</em>) – The folder to download the dataset to.</p></li>
<li><p><strong>version</strong> (<em>str</em>) – The version of the STRING database to download.</p></li>
<li><p><strong>taxon</strong> (<em>Optional</em><em>[</em><em>int</em><em>]</em>) – The NCBI taxon ID of the organism whose links you wish to download. Omit for all organisms. Defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download.string_physical_detailed_links">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">string_physical_detailed_links</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">version</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'12.0'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">taxon</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ppi_origami.__main__.Download.string_physical_detailed_links" title="Link to this definition">#</a></dt>
<dd><p>Download the <a class="reference external" href="https://string-db.org/">STRING</a> physical links dataset. Downloads the <code class="docutils literal notranslate"><span class="pre">protein.physical.links.detailed.{version}.txt.gz</span></code> file. This dataset only includes PPIs with evidence of physical interactions, and provide more information than <code class="docutils literal notranslate"><span class="pre">string_physical_links</span></code>.</p>
<p>You can call this function from the CLI using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">string_physical_detailed_links</span> <span class="n">RAW_FOLDER</span> <span class="o">--</span><span class="n">version</span> <span class="mf">12.0</span> <span class="o">--</span><span class="n">taxon</span> <span class="mi">9606</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>raw_folder</strong> (<em>pathlib.Path</em>) – The folder to download the dataset to.</p></li>
<li><p><strong>version</strong> (<em>str</em>) – The version of the STRING database to download.</p></li>
<li><p><strong>taxon</strong> (<em>Optional</em><em>[</em><em>int</em><em>]</em>) – The NCBI taxon ID of the organism whose links you wish to download. Omit for all organisms.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download.string_physical_links">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">string_physical_links</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">version</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'12.0'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">taxon</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ppi_origami.__main__.Download.string_physical_links" title="Link to this definition">#</a></dt>
<dd><p>Download the <a class="reference external" href="https://string-db.org/">STRING</a> physical links dataset. Downloads the <code class="docutils literal notranslate"><span class="pre">protein.physical.links.{version}.txt.gz</span></code> file. This dataset only includes PPIs with evidence of physical interactions.</p>
<p>You can call this function from the CLI using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">string_physical_links</span> <span class="n">RAW_FOLDER</span> <span class="o">--</span><span class="n">version</span> <span class="mf">12.0</span> <span class="o">--</span><span class="n">taxon</span> <span class="mi">9606</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>raw_folder</strong> (<em>pathlib.Path</em>) – The folder to download the dataset to.</p></li>
<li><p><strong>version</strong> (<em>str</em>) – The version of the STRING database to download.</p></li>
<li><p><strong>taxon</strong> (<em>Optional</em><em>[</em><em>int</em><em>]</em>) – The NCBI taxon ID of the organism whose links you wish to download. Omit for all organisms.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download.uniprot_delac">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">uniprot_delac</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ppi_origami.__main__.Download.uniprot_delac" title="Link to this definition">#</a></dt>
<dd><p>Download <a class="reference external" href="https://www.uniprot.org/">UniProt</a> deleted accessions to the raw folder. More info can be found in the <a class="reference external" href="https://www.uniprot.org/help/deleted_accessions">UniProtKB Manual</a>.</p>
<p>You can call this function from the CLI using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">uniprot_delac</span> <span class="n">RAW_FOLDER</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>raw_folder</strong> (<em>pathlib.Path</em>) – The folder to download the dataset to.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download.uniprot_id_mapping">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">uniprot_id_mapping</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ppi_origami.__main__.Download.uniprot_id_mapping" title="Link to this definition">#</a></dt>
<dd><p>Download <a class="reference external" href="https://www.uniprot.org/">UniProt</a> ID mappings to the raw folder. More info can be found on <a class="reference external" href="https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/idmapping/README">UniProt.org</a>.</p>
<p>You can call this function from the CLI using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">uniprot_id_mapping</span> <span class="n">RAW_FOLDER</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>raw_folder</strong> (<em>pathlib.Path</em>) – The folder to download the dataset to.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download.uniprot_sec_ac">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">uniprot_sec_ac</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ppi_origami.__main__.Download.uniprot_sec_ac" title="Link to this definition">#</a></dt>
<dd><p>Download UniProt secondary accessions to the raw folder.</p>
<p>You can call this function from the CLI using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">uniprot_sec_ac</span> <span class="n">RAW_FOLDER</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>raw_folder</strong> (<em>pathlib.Path</em>) – The folder to download the dataset to.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download.uniprot_seqs_db">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">uniprot_seqs_db</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">processed_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">taxon</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ppi_origami.__main__.Download.uniprot_seqs_db" title="Link to this definition">#</a></dt>
<dd><p>Download <a class="reference external" href="https://www.uniprot.org/">UniProt</a> sequences and saves them to a <a class="reference external" href="https://en.wikipedia.org/wiki/LevelDB">LevelDB</a> file in the processed folder. Will download sequences for the specified taxon if taxon is not <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<p>You can call this function from the CLI using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">uniprot_seqs_db</span> <span class="n">PROCESSED_FOLDER</span> <span class="o">--</span><span class="n">taxon</span> <span class="mi">9606</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>processed_folder</strong> (<em>pathlib.Path</em>) – The folder to save the database to.</p></li>
<li><p><strong>taxon</strong> (<em>int</em>) – The NCBI taxon ID of the organism whose links you wish to download. Omit to download sequences for all organisms. Defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download.uniprot_seqs_fasta">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">uniprot_seqs_fasta</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ppi_origami.__main__.Download.uniprot_seqs_fasta" title="Link to this definition">#</a></dt>
<dd><p>Download <a class="reference external" href="https://www.uniprot.org/">UniProt</a> sequences in the FASTA format to the raw folder.</p>
<p>You can call this function from the CLI using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">uniprot_seqs_fasta</span> <span class="n">RAW_FOLDER</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>raw_folder</strong> (<em>pathlib.Path</em>) – The folder to download the dataset to.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ppi_origami.__main__.Download.uniref">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">uniref</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threshold</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ppi_origami.__main__.Download.uniref" title="Link to this definition">#</a></dt>
<dd><p>Download the <a class="reference external" href="https://www.uniprot.org/uniref">UniRef</a> dataset from <a class="reference external" href="https://www.uniprot.org/">UniProt</a>. You must specify a similarity threshold among the three available options: 50%, 90%, and 100%.</p>
<p>You can call this function from the CLI using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_origami</span> <span class="n">download</span> <span class="n">uniref</span> <span class="n">RAW_FOLDER</span> <span class="n">THRESHOLD</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>raw_folder</strong> (<em>pathlib.Path</em>) – The folder to download the dataset to.</p></li>
<li><p><strong>threshold</strong> (<em>int</em>) – The UniRef identity threshold. Must be one of 50, 90, 100.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          <a class="next-page" href="process.html">
              <div class="page-info">
                <div class="context">
                  <span>Next</span>
                </div>
                <div class="title">Process</div>
              </div>
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
            </a>
          <a class="prev-page" href="../theory.html">
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">Theory</div>
                
              </div>
            </a>
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2024, Joseph Szymborski
            </div>
            Made with <a href="https://www.sphinx-doc.org/">Sphinx</a> and <a class="muted-link" href="https://pradyunsg.me">@pradyunsg</a>'s
            
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            On this page
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">Download</a><ul>
<li><a class="reference internal" href="#ppi_origami.__main__.Download"><code class="docutils literal notranslate"><span class="pre">Download</span></code></a><ul>
<li><a class="reference internal" href="#ppi_origami.__main__.Download.biogrid"><code class="docutils literal notranslate"><span class="pre">Download.biogrid()</span></code></a></li>
<li><a class="reference internal" href="#ppi_origami.__main__.Download.dscript"><code class="docutils literal notranslate"><span class="pre">Download.dscript()</span></code></a></li>
<li><a class="reference internal" href="#ppi_origami.__main__.Download.hippie"><code class="docutils literal notranslate"><span class="pre">Download.hippie()</span></code></a></li>
<li><a class="reference internal" href="#ppi_origami.__main__.Download.oma"><code class="docutils literal notranslate"><span class="pre">Download.oma()</span></code></a></li>
<li><a class="reference internal" href="#ppi_origami.__main__.Download.string_aliases"><code class="docutils literal notranslate"><span class="pre">Download.string_aliases()</span></code></a></li>
<li><a class="reference internal" href="#ppi_origami.__main__.Download.string_links"><code class="docutils literal notranslate"><span class="pre">Download.string_links()</span></code></a></li>
<li><a class="reference internal" href="#ppi_origami.__main__.Download.string_links_detailed"><code class="docutils literal notranslate"><span class="pre">Download.string_links_detailed()</span></code></a></li>
<li><a class="reference internal" href="#ppi_origami.__main__.Download.string_physical_detailed_links"><code class="docutils literal notranslate"><span class="pre">Download.string_physical_detailed_links()</span></code></a></li>
<li><a class="reference internal" href="#ppi_origami.__main__.Download.string_physical_links"><code class="docutils literal notranslate"><span class="pre">Download.string_physical_links()</span></code></a></li>
<li><a class="reference internal" href="#ppi_origami.__main__.Download.uniprot_delac"><code class="docutils literal notranslate"><span class="pre">Download.uniprot_delac()</span></code></a></li>
<li><a class="reference internal" href="#ppi_origami.__main__.Download.uniprot_id_mapping"><code class="docutils literal notranslate"><span class="pre">Download.uniprot_id_mapping()</span></code></a></li>
<li><a class="reference internal" href="#ppi_origami.__main__.Download.uniprot_sec_ac"><code class="docutils literal notranslate"><span class="pre">Download.uniprot_sec_ac()</span></code></a></li>
<li><a class="reference internal" href="#ppi_origami.__main__.Download.uniprot_seqs_db"><code class="docutils literal notranslate"><span class="pre">Download.uniprot_seqs_db()</span></code></a></li>
<li><a class="reference internal" href="#ppi_origami.__main__.Download.uniprot_seqs_fasta"><code class="docutils literal notranslate"><span class="pre">Download.uniprot_seqs_fasta()</span></code></a></li>
<li><a class="reference internal" href="#ppi_origami.__main__.Download.uniref"><code class="docutils literal notranslate"><span class="pre">Download.uniref()</span></code></a></li>
</ul>
</li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script src="../_static/documentation_options.js?v=5929fcd5"></script>
    <script src="../_static/doctools.js?v=888ff710"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/scripts/furo.js?v=32e29ea5"></script>
    </body>
</html>